### PR TITLE
revamp discord groups

### PR DESCRIPTION
**Pull request by pallabez**
Date: 20 04 2024

Developer Name: @pallabez 

---

## Issue Ticket Number
#732

## Description

Revamp the groups dashboard. Use /groups-dev to check it out

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
image
image
image

## Test Coverage
<details>
<summary>Screenshot 1</summary>

image

</details>

### CODE REVIEW

File: groups-dev/script.js [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+          roleId: group.roleid,
+          description: group.description,
+          isUpdating: false,
+        };
+        return acc;
+      }, {});
+      dataStore.discordId = roleData.userId;
+    })
+    .then(() => removeLoadingCards());
+};
```
Comment (ivinayakg) [inline groups-dev/script.js:146]: smart, 
can we have a better name for this, instead of when this function be called.

**Review by ivinayakg — COMMENTED** [review]
no tests.
I really liked the javascript here, saw some new things thanks.

File: groups-dev/constants.js
[the remark this answers is not in the document]
Comment (pallabez) [reply groups-dev/constants.js:1]: Thanks

File: groups-dev/script.js (hunk 1 above)
Comment (pallabez) [reply groups-dev/script.js:146]: I tried to replicate life cycle methods. onCreate, afterAuthentication came from that. 

What name would you suggest?

File: groups-dev/render.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+  function toggleSubmitButton({ enabled }) {
+    if (enabled) {
+      submitButton.classList.remove('submit__button--disabled');
+    } else {
+      submitButton.classList.add('submit__button--disabled');
+    }
+  }
+
+  const form = modalElement.querySelector('.form');
```
[the remark this answers is not in the document]
Comment (pallabez) [reply groups-dev/render.js:164]: We can but in this case, it is dynamically loaded so the querySelector may return empty if not careful.

File: groups-dev/createElements.js [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  return profileElement;
+};
+
+const createGroupCreationModal = (onClose = () => {}, onSubmit = () => {}) => {
+  const backdropElement = document.createElement('div');
+  const modalElement = document.createElement('div');
+  modalElement.className = 'group-creation-modal';
+  modalElement.innerHTML = `
+        <button type="button" id="close-button" class="button button--minimal">
+          <svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
```
Comment (palakgupta2712) [inline groups-dev/createElements.js:102]: can we save them as svg and then use them here?

File: groups-dev/createElements.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+  inputField.addEventListener('input', () => {
+    const inputValue = getInput();
+    const shouldDisableSubmit =
+      inputValue.length === 0 || inputValue.includes('group');
+
+    toggleSubmitButton({ enabled: !shouldDisableSubmit });
+  });
+
+  function toggleSubmitButton({ enabled }) {
```
Comment (palakgupta2712) [inline groups-dev/createElements.js:156]:
```
function toggleSubmitButton({ enabled }) {
  submitButton.classList.toggle('submit__button--disabled', !enabled);
}

```
you can use `toggle()` also here.

Comment (pallabez) [reply groups-dev/createElements.js:156]: Got it

File: groups-dev/render.js
[the remark this answers is not in the document]
Comment (pallabez) [reply groups-dev/render.js:1]: Done

File: groups-dev/createElements.js (hunk 3 above)
Comment (pallabez) [reply groups-dev/createElements.js]: Done

**Review by palakgupta2712 — APPROVED** [review]

**Review by Achintya-Chatterjee — COMMENTED** [review]
@pallabez convert all of these SVG's and  PNG's  into webP

File: groups-dev/script.js (hunk 1 above)
Comment (ivinayakg) [reply groups-dev/script.js]: that is interesting, great then 👍
any blog on these lifecycle methods?

**Pull request merged**